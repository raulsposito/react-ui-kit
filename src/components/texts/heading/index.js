import React from 'react'
import PropTypes from 'prop-types'

import MyPropTyes from '@global/propTypes'

import { H1, H2, H3, H4, H5, H6 } from './styles'

const Heading = ({ weight, size, text, type, align, color, children, marginBottom, disabled }) => {
  const Tag = (() => {
    switch (type) {
      case 'h2': return H2
      case 'h3': return H3
      case 'h4': return H4
      case 'h5': return H5
      case 'h6': return H6
      default: return H1
    }
  })()

  return (
    <Tag
      weight={weight}
      color={color}
      align={align}
      size={size}
      disabled={disabled}
      marginBottom={marginBottom}
    >
      {text || children}
    </Tag>
  )
}

Heading.propTypes = {
  text: PropTypes.string,
  type: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']),
  align: PropTypes.oneOf(['left', 'center', 'right']),
  color: MyPropTyes.color,
  weight: MyPropTyes.weight,
  size: MyPropTyes.size,
  children: PropTypes.node,
  marginBottom: MyPropTyes.space,
  disabled: PropTypes.bool
}

export default Heading
