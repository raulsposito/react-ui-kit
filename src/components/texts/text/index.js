import React from 'react'
import PropTypes from 'prop-types'

import MyPropTyes from '@global/propTypes'

import { StyledText } from './styles'

const Text = ({ text, align, weight, color, size, children, marginBottom, disabled }) => (
  <StyledText weight={weight} size={size} color={color} align={align} marginBottom={marginBottom} disabled={disabled}>
    {text || children}
  </StyledText>
)

Text.propTypes = {
  text: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  align: PropTypes.oneOf(['left', 'center', 'right']),
  weight: MyPropTyes.weight,
  color: MyPropTyes.color,
  size: MyPropTyes.size,
  children: PropTypes.node,
  marginBottom: MyPropTyes.space,
  disabled: PropTypes.bool
}

export default Text
