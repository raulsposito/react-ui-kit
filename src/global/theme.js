// https://www.color-hex.com/color/8293ee

export const colors = {
  primary: {
    regular: '#FC760D',
    dark: '#E46B0B',
    darker: '#C66F2B',
    hover: '#E17622',
    focus: '#B97641',
    selected: '#D4762C',
    disabled: '#F99546'
  },
  variant: {
    regular: '#FCB10D',
    dark: '#E29F0C',
    darker: '#C9A557',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  secondary: {
    regular: '#2FCFCF',
    dark: '#1F9A9A',
    darker: '#147676',
    hover: '#3DBDBD',
    focus: '#52A2A2',
    selected: '#44B4B4',
    disabled: '#5FD8D8'
  },
  success: {
    regular: '#3DB265',
    dark: '#48A668',
    darker: '#59946D',
    hover: '#48A668',
    focus: '#59946D',
    selected: '#59946D',
    disabled: '#9EBCA8'
  },
  error: {
    regular: '#FF5668',
    dark: '#E45C6B',
    darker: '#BA666F',
    hover: '#E45C6B',
    focus: '#BA666F',
    selected: '#BA666F',
    disabled: '#F8A4AC'
  },
  warning: {
    regular: '#FFAF00',
    dark: '#E5BB4D',
    darker: '#CCA644',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  contrast: {
    regular: '#3C3C3C',
    disabled: '#E4E4E4'
  },
  system: {
    regular: '#BAB5B4',
    dark: '#CBC6C4',
    darker: '#D7D4D1',
    hover: '#ACA8A8',
    focus: '#989595',
    selected: '#A6A2A1',
    disabled: '#C8C4C3'
  },
  background: {
    primary: '#FFFFFF',
    regular: '#FFFFFF',
    hover: '#F1F1F1',
    disabled: '#F8F8F8'
  },
  events: {
    inprogress: '#3DB265',
    next: '#5678CF',
    finished: '#BAB5B4',
    suspended: '#FFD056',
    cancelled: '#FF5668',
    white: '#FFFFFF'
  }
}

export const spacing = {
  quarter: '2px',
  half: '4px',
  threeQuarters: '6px',
  one: '8px',
  oneAndAQuarter: '10px',
  oneAndAHalf: '12px',
  two: '16px',
  twoAndAQuarter: '18px',
  twoAndAHalf: '20px',
  three: '24px',
  threeAndAHalf: '28px',
  four: '32px',
  fourAndAHalf: '36px',
  five: '40px',
  fiveAndAHalf: '44px',
  six: '48px',
  sixAndAHalf: '52px',
  seven: '56px',
  sevenAndAHalf: '60px',
  eight: '64px',
  eightAndAHalf: '68px',
  nine: '72px',
  nineAndAHalf: '76px',
  ten: '80px',
  tenAndAHalf: '84px',
  eleven: '88px',
  elevenAndAHalf: '92px',
  twelve: '96px',
  twelveAndAHalf: '100px',
  thirteen: '104px',
  fourteen: '112x',
  fifteen: '120px'
}

export const page = {
  width: '1200px'
}
