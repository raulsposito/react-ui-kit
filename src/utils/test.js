export const getComponents = styled => {
  const componentClass = styled.styledComponentId
  return document.getElementsByClassName(componentClass)
}

export const getComponent = (styled, index = 0) => {
  const components = getComponents(styled)
  return components[index]
}
