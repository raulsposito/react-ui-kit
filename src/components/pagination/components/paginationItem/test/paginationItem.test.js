import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { colors, spacing } from '@global/theme'
import { getComponent } from '@utils/test'

import { Text } from '@components/texts'
import Icon from '@components/icon'

import PaginationItem from '..'

const mockData = {
  number: 12,
  icon: 'first_page',
  onClick: jest.fn()
}

describe('<PaginationItem />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<PaginationItem number={mockData.number} onClick={mockData.onClick} />)
    expect(asFragment()).toMatchSnapshot()
  })

  describe('Child Element', () => {
    it('has number', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} />)
      const number = screen.queryByText(mockData.number)
      const icon = getComponent(Icon)

      expect(number).toBeInTheDocument()
      expect(icon).toBeUndefined()
    })

    it('has icon', () => {
      render(<PaginationItem icon={mockData.icon} onClick={mockData.onClick} />)
      const number = getComponent(Text)
      const icon = screen.queryByText(mockData.icon)

      expect(number).toBeUndefined()
      expect(icon).toBeInTheDocument()
    })
  })

  describe('Sizes', () => {
    it('has default size', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveStyle({ height: `${spacing.fourAndAHalf}` })
      expect(paginationItem).toHaveStyle({ width: `${spacing.fourAndAHalf}` })
    })

    it('has small size', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} size='small' />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveStyle({ height: `${spacing.threeAndAHalf}` })
      expect(paginationItem).toHaveStyle({ width: `${spacing.threeAndAHalf}` })
    })

    it('has medium size', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} size='medium' />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveStyle({ height: `${spacing.fourAndAHalf}` })
      expect(paginationItem).toHaveStyle({ width: `${spacing.fourAndAHalf}` })
    })

    it('has large size', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} size='large' />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveStyle({ height: `${spacing.five}` })
      expect(paginationItem).toHaveStyle({ width: `${spacing.five}` })
    })
  })

  describe('Disabled', () => {
    it('is disabled', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} disabled />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveAttribute('disabled')
      expect(paginationItem).toHaveStyle({ backgroundColor: `${colors.background.regular}` })
      expect(paginationItem).toHaveStyle({ cursor: 'default' })
      expect(paginationItem).toHaveStyle({ pointerEvents: 'none' })
    })

    it('is not disabled', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveStyle({ backgroundColor: `${colors.background.regular}` })
      expect(paginationItem).toHaveStyle({ cursor: 'pointer' })
    })
  })

  describe('Selected', () => {
    it('is selected', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} selected />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveStyle({ backgroundColor: `${colors.primary.selected}` })
    })

    it('is selected & disabled', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} selected disabled />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveAttribute('disabled')
      expect(paginationItem).toHaveStyle({ backgroundColor: `${colors.primary.disabled}` })
      expect(paginationItem).toHaveStyle({ cursor: 'default' })
      expect(paginationItem).toHaveStyle({ pointerEvents: 'none' })
    })

    it('is not selected', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveStyle({ backgroundColor: `${colors.background.regular}` })
      expect(paginationItem).toHaveStyle({ cursor: 'pointer' })
    })
  })

  describe('Colors', () => {
    it('has default color', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} selected />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveStyle({ backgroundColor: `${colors.primary.selected}` })
    })

    it('is primary', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} color='primary' selected />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveStyle({ backgroundColor: `${colors.primary.selected}` })
    })

    it('is variant', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} color='variant' selected />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveStyle({ backgroundColor: `${colors.variant.selected}` })
    })

    it('is secondary', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} color='secondary' selected />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveStyle({ backgroundColor: `${colors.secondary.selected}` })
    })

    it('is success', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} color='success' selected />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveStyle({ backgroundColor: `${colors.success.selected}` })
    })

    it('is error', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} color='error' selected />)
      const paginationItem = screen.getByRole('button')

      expect(paginationItem).toHaveStyle({ backgroundColor: `${colors.error.selected}` })
    })
  })

  describe('Mouse Events', () => {
    it('calls function on click', () => {
      render(<PaginationItem number={mockData.number} onClick={mockData.onClick} />)
      const paginationItem = screen.getByRole('button')

      userEvent.click(paginationItem)
      expect(mockData.onClick).toHaveBeenCalledTimes(1)
    })
  })
})
