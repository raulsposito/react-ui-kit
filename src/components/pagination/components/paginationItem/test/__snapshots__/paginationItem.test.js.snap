// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationItem /> should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-jrAFXE jqYNa-D"
  >
    <p
      class="sc-bdfBQB iDqXfG"
      color="contrast"
    >
      12
    </p>
  </button>
</DocumentFragment>
`;
